e2eTest/src/main/java/com/google/android/ground/e2etest/SurveyRunnerTest.kt

```diff
@@ -0,0 +1,235 @@
+ * limitations under the License.
+ */
+package com.google.android.ground.e2etest
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.By
+import androidx.test.uiautomator.Direction
+import androidx.test.uiautomator.UiDevice
+import androidx.test.uiautomator.Until
+import com.google.android.ground.R
+import com.google.android.ground.e2etest.TestConfig.GROUND_PACKAGE
+import com.google.android.ground.e2etest.TestConfig.LONG_TIMEOUT
+import com.google.android.ground.e2etest.TestConfig.SHORT_TIMEOUT
+import com.google.android.ground.e2etest.TestConfig.TEST_SURVEY_IDENTIFIER
+import com.google.android.ground.e2etest.TestConfig.TEST_SURVEY_TASKS_ADHOC
+import com.google.android.ground.model.task.Task
+import junit.framework.TestCase.fail
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class SurveyRunnerTest : AutomatorRunner {
+
+  override lateinit var device: UiDevice
+
+  private val selectSurveyTitle = stringResource(R.string.select_survey_title)
+  private val collectData = stringResource(R.string.collect_data)
+  private val newDataCollectionSite = stringResource(R.string.new_site)
+  private val close = stringResource(R.string.close)
+  private val dropPin = stringResource(R.string.drop_pin)
+  private val addPoint = stringResource(R.string.add_point)
+  private val complete = stringResource(R.string.complete_polygon)
+  private val capture = stringResource(R.string.capture)
+  private val camera = stringResource(R.string.camera)
+  private val ok = stringResource(R.string.ok)
+  private val next = stringResource(R.string.next)
+  private val done = stringResource(R.string.done)
+  private val save = stringResource(R.string.save)
+
+  @Test
+  fun run() {
+    launchGround()
+    signIn()
+    selectTestSurvey()
+    zoomIntoLocation()
+    startAdHocLoiTask()
+    fillOutTaskData(isAdHoc = true, TEST_SURVEY_TASKS_ADHOC)
+    clickSubmissionConfirmationDone()
+    startPredefinedLoiTask()
+    fillOutTaskData(isAdHoc = false, TEST_SURVEY_TASKS_ADHOC.drop(1))
+    clickSubmissionConfirmationDone()
+  }
+
+  private fun launchGround() {
+    // Initialize UiDevice instance.
+    device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+
+    launchPackage(GROUND_PACKAGE)
+
+    // Wait for the app to appear.
+    if (!device.wait(Until.hasObject(By.pkg(GROUND_PACKAGE).depth(0)), LONG_TIMEOUT)) {
+      fail("Failed to launch app.")
+    }
+  }
+
+  private fun signIn() {
+    if (!waitClickGone(By.clazz("android.widget.Button"), LONG_TIMEOUT)) {
+      fail("Failed to sign in.")
+    }
+  }
+
+  private fun selectTestSurvey() {
+    device.wait(Until.hasObject(By.text(selectSurveyTitle)), LONG_TIMEOUT)
+    val testSurveySelector =
+      By.clazz("androidx.cardview.widget.CardView")
+        .hasDescendant(By.clazz("android.widget.TextView").textContains(TEST_SURVEY_IDENTIFIER))
```

Comment:
Amazing! I had to tweak to: `fun <T : Any> byClass(kclass: KClass<T>): BySelector = By.clazz(kclass.java.name)` since the direct to class version wasn't working.